### PR TITLE
Added SelectedItem component; refactored tests

### LINKED ISSUE #35: Separating the current skill displayer into its own component
Right now we're displaying the current skill's information with static elements (down at the bottom of the Skills.jsx file). We could make a separate component to decouple the current skill's interface from the general Skill component.

### PR DESCRIPTION
**Pull request by solkaz**
Fixes #35 and #31 

Adds a `SelectedItem` component that handles interactions that require a selected item (deletion/add links)

### CODE REVIEW

**Comment by thebho** [comment]
Waiting on merge conflicts

**Comment by solkaz** [comment]
I've rebased my branch, and the tests passed, but there was something fishy:

```
npm test


> skilldirectory@0.1.0 test /home/ubuntu/skilldirectoryui
> react-scripts test --env=jsdom

 PASS  src/Skills.test.jsx
 PASS  src/LinksForm.test.jsx
 PASS  src/Team.test.jsx
 PASS  src/DeleteModal.test.jsx
 PASS  src/SkillsForm.test.jsx
 PASS  src/Header.test.jsx
 PASS  src/Home.test.jsx
 PASS  src/App.test.jsx

Test Suites: 8 passed, 8 total
Tests:       33 passed, 33 total
Snapshots:   8 passed, 8 total
Time:        7.636s
Ran all test suites.
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 7): Error: Network Error
(node:16668) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): Error: Network Error
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 9): Error: Network Error
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 10): Error: Network Error
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 11): Error: Network Error
(node:16668) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 12): Error: Network Error
```

**Review by thebho — APPROVED** [review]

**Comment by thebho** [comment]
More conflicts, sorry Jeff

File: src/Skills.jsx [hunk 1]
```diff
@@ -193,19 +233,14 @@ class Skills extends Component {
...
-
             <Modal
               isOpen={this.state.skillModalIsOpen}
               onRequestClose={this.closeSkillModal}
               contentLabel="SkillModal"
-              style={ModalStyle}
             >
               <SkillForm api={api}
-                         closeModal={this.closeSkillModal}
-                         reloadSkills={this.reloadSkills}
```
Comment (thebho) [inline src/Skills.jsx:205]: Is this logic moved somewhere?

**Review by thebho — APPROVED** [review]

File: src/Skills.jsx (hunk 1 above)
Comment (solkaz) [reply src/Skills.jsx:205]: CloseSkillModal is still passed to `SkillForm` but `reloadSkills` will not be, as it isn't called anywhere in `SkillForm` anyways.

Comment (thebho) [reply src/Skills.jsx:205]: So when you add a skill does it not reload the skills dropdown?

Comment (solkaz) [reply src/Skills.jsx:205]: It doesn't in this PR, but I address that in [this PR](https://github.com/maryvilledev/skilldirectoryui/pull/86)

**Pull request merged**